lib/src/sliver_infinite_list.dart

```diff
@@ -156,19 +78,9 @@ class _SliverInfiniteListInternalState
   }
 
   @override
-  void didChangeDependencies() {
-    super.didChangeDependencies();
-
-    detachFromPosition();
-    scrollPosition = Scrollable.of(context)?.position;
-    attachToPosition();
-  }
-
-  @override
-  void didUpdateWidget(_SliverInfiniteListInternal oldWidget) {
+  void didUpdateWidget(SliverInfiniteList oldWidget) {
     super.didUpdateWidget(oldWidget);
-    if (widget.itemCount != oldWidget.itemCount ||
-        widget.hasReachedMax != oldWidget.hasReachedMax) {
+    if (widget.hasReachedMax != oldWidget.hasReachedMax) {
```

Comment:
What if the `itemCount` has changed? Don't we still want to attempt a fetch?

Reply:
not directly, now that our fetch happens due to layout changes, changing this count should trigger fetch if the new items would affect the viewport.

Reply:
Ah got it. So in this case if `hasReachedMax` has changed from `false` to `true` isn't there no point in attempting a fetch? It doesn't really matter to much because `attemptFetch` has a check for this but might be good to only add the `postFrameCallback` if `hasReachedMax` has gone from true to false wdyt?

Reply:
Makes sense.

Reply:
Since every call to `attemptFetch` was on a frame callback, moved the scheduling to inside it (inside the if too).